test/elixir/test/disk_monitor.exs

```diff
@@ -0,0 +1,58 @@
+defmodule DiskMonitorTest do
+  use CouchTestCase
+
+  @moduletag :disk_monitor
+
+  setup_all do
+    set_config({"disk_monitor", "enable", "true"})
+    #on_exit(fn ->
+    #  set_config({"disk_monitor", "enable", "false"})
+    #end
```

Comment:
This automatically gets reset by the test runner? Should we remove the comments then?

Reply:
oof sorry. I forget this. I can't get on_exit to work anywhere at all, I just get an error saying it can only be called from the test function. It doesn't work when I call it there or in setup/setup_all or anywhere at all. I think it's broken somehow.

Reply:
looks like a bug, or at least a very poor error message, in ExUnit.

Reply:
removed this section. as you say set_config adds its own on_exit. doing an on_exit from inside an on_exit is why I was 'not in the test handler function' despite appearances.